robot/impl/local_robot.go

```diff
@@ -1207,6 +1207,9 @@ func (r *localRobot) reconfigure(ctx context.Context, newConfig *config.Config,
 			} else {
 				canReconfigure, err := r.checkMaintenanceSensorReadings(ctx, newConfig.MaintenanceConfig.MaintenanceAllowedKey, sensorComponent)
 				if !canReconfigure {
+					if err != nil {
+						r.logger.CErrorw(ctx, "error reading maintenance sensor", "error", err)
+					}
 					r.logger.Info("maintenance_allowed_key found from readings on maintenance sensor. Skipping reconfiguration.")
```

Comment:
we should only log once here, put this in an if?

Reply:
putting it in an else so it either logs the error or the skipping log